lib/kernel/src/erl_boot_server.erl

```diff
@@ -141,6 +189,21 @@ check_arg([], Result) ->
 check_arg(_, _Result) ->
     error.
 
+check_options(Options) when map_size(Options) =:= 0 ->
+    ok;
+check_options(Options) ->
+    if
+        is_map_key(listen_port, Options),
+        map_size(Options) =:= 1 ->
+            ok;
+        true ->
+            error
+    end.
```

Comment:
Checking `map_size` is a bit awkward when all we want to do is ensure that all given pairs are valid options.

Perhaps do something like this instead?

```suggestion
check_options(Options) when is_map(Options) ->
    lists:all(fun valid_option/1, maps:to_list(Options));
check_options(_) ->
    false.

valid_option({listen_port, Port}) when is_integer(Port) -> true;
valid_option({_, _}) -> false.
```